zulip_bots/zulip_bots/bots/googlesearch/test_googlesearch.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import random
+
+from zulip_bots.test_lib import BotTestCase
+
+class TestGoogleSearchBot(BotTestCase):
+    bot_name = 'googlesearch'
+
+    # Simple query
+    def test_bot(self):
+        self.assert_bot_response({'content': 'zulip'}, {'content': 'Success: https://zulipchat.com/'}, 'send_reply')
+
+    # Help without typing anything
+    def test_bot_help_none(self):
+        help_message = "To use this bot, start messages with @mentioned-bot, \
+                    followed by what you want to search for. If \
+                    found, Zulip will return the first search result \
+                    on Google.\
+                    \
+                    An example message that could be sent is:\
+                    '@mentioned-bot zulip' or \
+                    '@mentioned-bot how to create a chatbot'."
+        self.assert_bot_response({'content': ''}, {'content': help_message}, 'send_reply')
+
+    # Help from typing 'help'
+    def test_bot_help(self):
+        help_message = "To use this bot, start messages with @mentioned-bot, \
+                    followed by what you want to search for. If \
+                    found, Zulip will return the first search result \
+                    on Google.\
+                    \
+                    An example message that could be sent is:\
+                    '@mentioned-bot zulip' or \
+                    '@mentioned-bot how to create a chatbot'."
+        self.assert_bot_response({'content': 'help'}, {'content': help_message}, 'send_reply')
+
+    # A google search with no results
+    def test_bot_no_results(self):
+        ranstr = ''
+        characters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890'
+        for i in range(0, 100):
+            ranstr += characters[random.randrange(len(characters))]
+        self.assert_bot_response({'content': ranstr}, {'content': 'Error: Search failed.'}, 'send_reply')
```

Comment:
I don't see any code where you are mocking network requests. I think that the giphy bot's tests should provide you a good example of how that should work.